Scripts/DowloadPortalItemData.swift

```diff
@@ -244,33 +245,43 @@ var downloadedItems = previousDownloadedItems
 // Asynchronously downloads portal items.
 let dispatchGroup = DispatchGroup()
 
-portalItems.forEach { portalItem in
+for portalItem in portalItems {
+    // Checks to see if an item is already downloaded.
+    guard downloadedItems[portalItem.identifier] == nil else {
+        print("note: Item already downloaded: \(portalItem.identifier)")
+        continue
+    }
+    
     let destinationURL = downloadDirectoryURL.appendingPathComponent(portalItem.identifier, isDirectory: true)
-    // Checks if a directory exists or not, to see if an item is already downloaded.
-    if FileManager.default.fileExists(atPath: destinationURL.path) {
-        print("info: Item \(portalItem.identifier) has already been downloaded.")
-    } else {
-        do {
-            // Creates an enclosing directory with portal item ID as its name.
-            try FileManager.default.createDirectory(at: destinationURL, withIntermediateDirectories: false)
-        } catch {
-            print("error: Error creating download directory: \(error.localizedDescription).")
+    
+    // Deletes the directory if it already exists.
+    // This happens when the item is not in the plist and needs to be redownloaded.
+    do {
+        try FileManager.default.removeItem(at: destinationURL)
+        print("note: Deleted directory: \(portalItem.identifier)")
+    }
```

Comment:
We don't want to throw errors from the script because it gives an unsightly stack dump:

![image](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/2257493/ff2b6841-e569-47fa-b181-0bd4a5b3e4e4)

That's why elsewhere we catch any error and explicitly exit with a code of `1`. Also elsewhere, we only print a message for unexpected things (eg. errors). I would say that this print statement is unnecessary and should be removed. And as I said before, if `removeItem(at:)` does throw an error, it is almost certainly because the item (directory) doesn't exist, so we can just move on. No need to try to handle that error, which is why I suggested to just use `try?`.